app/app.js

```diff
@@ -120,15 +125,143 @@ let Marvel = {
         showFirstLast: false
       })
     );
+  },
+  addRefinementList(){
+    this.search.addWidget(
+      instantsearch.widgets.currentRefinedValues({
+        container: '#current-refined-values',
+        clearAll: 'after'
+      })
+    );
+  },
+  renderHits(){
+    var renderedHits = {
+      render: function(options) {
+        processHeroProfile();
```

Comment:
That is my biggest problem regarding rebinding sicne I'm using Algolia Instantsearch. 
Never certain of what could be the best way to do this. I'll be glad to see how you would manage it!